`lscpu`: Support extracting and outputting nested fields

alxndrv opened the pull request:
`lscpu` should be capable of outputting nested information. When using human-readable output this is achieved simply using indentation, where as in JSON mode nested fields go under the `children` property.

### Core changes:
- Modified `CpuInfo` to be a recursive structure, allowing for nesting of entries through the `children` property.
- Moved output logic into its own function, adding support for printing out nested values, and ensuring the values on the right are aligned to the same column.
- Implemented a helper function for extracting arbitrary values from `/proc/cpuinfo`, using a dynamically constructed Regex.
- Added support for some basic fields like Vendor ID, CPU Family and Model, to illustrate how the aforementioned changes can be used in action.
- Added missing `-J` shorthand counterpart to the `--json` option.

### Output:
```
CPU(s):          16
Architecture:    x86_64
  Address sizes: 48 bits physical, 48 bits virtual
  Byte Order:    Little Endian
Vendor ID:       AuthenticAMD
  Model name:    AMD Ryzen 7 7700X 8-Core Processor
    CPU Family:  25
    Model:       97
```

cakebaker commented on src/uu/lscpu/src/lscpu.rs:
```diff
@@ -49,41 +61,145 @@ impl CpuInfos {
-        if let Some(cap) = re.captures_iter(&contents).next() {
-            cpu_infos.push("Model name", &cap[1]);
-        };
+    cpu_infos.push(CpuInfo::new(
+        "CPU(s)",
+        &format!("{}", system.cpus().len()),
+        None,
+    ));
```
The original `lscpu` shows the CPU information after the architecture information and so I would move this block down.

cakebaker commented on src/uu/lscpu/src/lscpu.rs:
```diff
@@ -49,41 +61,145 @@ impl CpuInfos {
+        cmp::max(own_width, max_child_width)
+    }
+
+    // Used later to align all values to the same column
+    let max_field_width = infos
+        .lscpu
+        .iter()
+        .map(|info| get_max_field_width(info, 0))
+        .max()
+        .unwrap();
```
It's a bit confusing to have code between two functions and I would move it down.

cakebaker reviewed (commented):
The rest looks good. Some tests would be nice ;-)

alxndrv commented:
> The rest looks good. Some tests would be nice ;-)

Thanks for the review. Moved the code around a bit as you suggested, and added a basic test for non-JSON output. One thing to note is that with these changes, `lscpu` becomes explicitly dependent on Linux, since the tests expect that we can read information from `/sys/devices/system/cpu` and `/proc/cpuinfo`, which aren't available on other systems. 

I've gone ahead and added flags to the tests so they only get executed on Linux builds. Let me know if you have better ideas for a long-term approach :slightly_smiling_face:

cakebaker commented:
Thanks!

> I've gone ahead and added flags to the tests so they only get executed on Linux builds. Let me know if you have better ideas for a long-term approach

The long-term approach is probably to use a crate like `sysinfo` that supports multiple platforms.
